ppr-ui/tests/unit/ConfirmDischarge.spec.ts

```diff
@@ -116,8 +118,20 @@ describe('ConfirmDischarge registration view', () => {
   })
 
   it('processes submit button action', async () => {
+    // Use mock service directly - account id can be anything.
+    const currentAccount = {
+      id: 'test_id'
+    }
+    sessionStorage.setItem('CURRENT_ACCOUNT', JSON.stringify(currentAccount))
+    sessionStorage.setItem('PPR_API_URL', 'https://bcregistry-bcregistry-mock.apigee.net/mockTarget/ppr/api/v1/')
+    // Set up for valid discharge request
+    await store.dispatch('setRegistrationNumber', '023001B')
+    await store.dispatch('setFolioOrReferenceNumber', 'A-00000402')
+    await store.dispatch('setRegistrationConfirmDebtorName', mockedDebtorNames[0])
+
     await wrapper.findComponent(DischargeConfirmSummary).vm.$emit('valid', true)
     await wrapper.findComponent(ButtonsStacked).vm.$emit('submit', true)
-    // fill in with the rest of the flow once built
+    await flushPromises()
+    expect(wrapper.vm.$route.name).toBe(RouteNames.CONFIRM_DISCHARGE)
```

Comment:
shouldn't this have routed to the dashboard if it was successful? To ensure the mock url call isn't failing you could mock the response via sinon if you want

Reply:
Yes, you are right, the call is failing in the test and staying at the confirm page.

Reply:
Unit test updated, now routes to the dashboard.